ODIN_II/SRC/soft_logic_def_parser.cpp

```diff
@@ -0,0 +1,173 @@
+	//else keep as is and try to open it
+
+	FILE *input_file = fopen(global_args.adder_def.value().c_str(),"r");
+	if(input_file)
+	{
+		printf("Reading soft_logic definition file @ %s ... ", global_args.adder_def.value().c_str());
+
+		soft_def_map[std::string("+_0")] = NULL;
+		soft_def_map[std::string("/_0")] = NULL;
+		soft_def_map[std::string("%_0")] = NULL;
+		soft_def_map[std::string("<<_0")] = NULL;
+
+		int error = 0;
+		int line_number = 0;
+		char line_buf[1024] = { 0 };
+		while (fgets(line_buf, 1024, input_file) != NULL && !error)
+		{
+			std::string line = line_buf;
+			line.erase(std::remove(line.begin(), line.end(), '\n'), line.end());
+			line_number += 1;
+
+			if(line == "")
+				break;
+			
+			std::vector<std::string> tokens;
+			char *line_dup = vtr::strdup(line.c_str());
+			char *temp_str = strtok(line_dup,",");
+			while(1)
+			{				
+				if(!temp_str)
+				{
+					error = (tokens.size() == 5)? 0: 1;
+					break;
+				}
+				tokens.push_back(std::string(temp_str));
+				temp_str = strtok(NULL,",");
+			}
+
+			if(!error)
+			{
+				std::string operation_name 			= tokens[0];
+				int operation_bitsize			 			=	std::stoi(tokens[1]);
+				std::string soft_hard						= tokens[2];
+				std::string sub_structure_name	= tokens[3];
+				int sub_structure_bitsize				=	std::stoi(tokens[4]);
+
+				std::string lookup = operation_name + "_0";
+
+				auto candidate = soft_def_map.find(lookup);
+				if(candidate == soft_def_map.end() \
+				|| operation_bitsize < 1 \
+				|| operation_bitsize < sub_structure_bitsize \
+				|| (soft_hard != "hard" && soft_hard != "soft"))
+				{
+					error =1;
+					vtr::free(line_dup);
+					break;
+				}
+				else
+				{
+					std::string key_map = operation_name + "_" + tokens[1];
+					soft_sub_structure* def = (soft_sub_structure*)vtr::malloc(sizeof(soft_sub_structure));
+					def->type= vtr::strdup(soft_hard.c_str());
+					def->name= vtr::strdup(sub_structure_name.c_str());
+					def->bitsize = sub_structure_bitsize;
+
+					soft_def_map[key_map] = def;
+				}
+			}
+			vtr::free(line_dup);
+		}
+		fclose(input_file);
+
+		if(error)
+		{
+			/* something went wrong! empty everything and throw a warning */
+			soft_def_map = std::map<std::string,soft_sub_structure*>();
+			warning_message(PARSE_ERROR, -1, -1, "ERROR parsing adder soft logic definition file @ line::%d\n",line_number);
+		}
+
+  	}
+}
+
+/*---------------------------------------------------------------------------------------------
+ * gets the soft blk size at a certain point in the adder using a define file input
+ *-------------------------------------------------------------------------------------------*/
+soft_sub_structure *fetch_blk(std::string op, int width)
+{
+	std::string lookup = op + "_" + std::to_string(width);
+
+	auto candidate = soft_def_map.find(lookup);
+	if(candidate == soft_def_map.end())
+		return NULL;
+
+	return soft_def_map[lookup];
+}
```

Comment:
drop this file since you don't need to parse an input file for the adder definition